docs/docs/api-reference/refine/Primitive_Checkers.md

```diff
@@ -0,0 +1,279 @@
+
+Validates a value as a `boolean`
+
+```jsx
+// define checker
+const check = boolean();
+
+// test a value
+const result = check(false);
+assert(result.type === 'success');
+
+// result should typecheck
+const value: boolean = result.value;
+
+// test an invalid value
+const failedResult = check(1);
+assert(failedResult.type === 'failure');
+```
+
+## `number()`
+
+Validates a value as a `number`
+
+```jsx
+// define checker
+const check = number();
+
+// test a value
+const result = check(1);
+assert(result.type === 'success');
+
+// result should typecheck
+const value: number = result.value;
+
+// test an invalid value
+const failedResult = check(false);
+assert(failedResult.type === 'failure');
+```
+
+## `string()`
+
+Validates a value as a `string`
+
+```jsx
+// define checker
+const check = string();
+
+// test a value
+const result = check('test');
+assert(result.type === 'success');
+
+// result should typecheck
+const value: string = result.value;
+
+// test an invalid value
+const failedResult = check(false);
+assert(failedResult.type === 'failure');
+```
+
+`string` can also take in a regex argument for validation.
+
+```jsx
+// define checker
+const check = string(/^users?$/);
+
+// test a value
+const result = check('user');
+assert(result.type === 'success');
+
+// result should typecheck
+const value: string = result.value;
+
+// test an invalid value
+const failedResult = check('buser');
+assert(failedResult.type === 'failure');
+```
+
+
+## `literal()`
+
+Validates a value as a given literal type
+
+```jsx
+// define checker
+// note: to get Flow to use the literal, we must annotate
+const check = literal<'add_todo'>('add_todo');
+
+// can also use for null/undefined/true/false literals
+const checkExactlyNull = literal<null>(null);
+
+// test a value
+const result = check('add_todo');
+assert(result.type === 'success');
+
+// result should typecheck
+const value: 'add_todo' = result.value;
+
+// test an invalid value
+const failedResult = check('remove_todo');
+assert(failedResult.type === 'failure');
+```
+
+## `stringLiterals()`
+
+Checker to assert if a mixed value matches a union of string literals.
+Legal values are provided as key/values in an object and may be translated by
+providing different values in the object.
+
+```jsx
+const suitChecker = stringLiterals({
+  heart: 'heart',
+  spade: 'spade',
+  club: 'club',
+  diamond: 'diamond',
+});
+
+const suit: 'heart' | 'spade' | 'club' | 'diamond' = assertion(suitChecker())(x);
+```
+
+## `date()`
+
+Validates a value as a javascript `Date` object
+
+```jsx
+// define checker
+const check = date();
+
+// test a value
+const result = check(new Date());
+assert(result.type === 'success');
+
+// result should typecheck
+const value: Date = result.value;
+
+// test an invalid value
+const failedResult = check(1);
+assert(failedResult.type === 'failure');
+```
+
+## `jsonDate()`
+
+Similar to date, though also will implicitly coerce ISO date strings to Date objects.
```

Comment:
Probably useful to mention that this is particularly useful for synchronizing with JSON serialization of Date objects